src/resolve-dependency.ts

```diff
@@ -1,42 +1,72 @@
 import { isAbsolute, resolve, sep } from 'path';
 import { Job } from './node-file-trace';
 
+export type FilesToEmit = Array<{
+  file: string,
+  type: string,
+  parent?: string,
+  isRealpath?: boolean
+}>
+
 // node resolver
 // custom implementation to emit only needed package.json files for resolver
 // (package.json files are emitted as they are hit)
 export default async function resolveDependency (specifier: string, parent: string, job: Job, cjsResolve = true): Promise<string | string[]> {
+  const cacheKey = specifier + parent
```

Comment:
Nice improvement, I'm sure this saves a lot of work.

The cache key should take into account the environment conditions as well here. Also it's advisable to properly terminate. Eg something like:

```js
const { conditions } = job;
const cacheKey = JSON.stringify({ specifier, parent, conditions })
```

Reply:
Good catch, updated to include that